app/jobs/email_privacy_policy_update_job.rb

```diff
@@ -0,0 +1,55 @@
+class EmailPrivacyPolicyUpdateJob < ApplicationJob
+  def perform(petition, limit: 1000)
+    terminating = false
+
+    worker = trap("TERM") do
+      terminating = true
+      worker.call
```

Comment:
NB This logic is copied from existing jobs, and should reschedule the job is the worker resets. But I'd be happier if I could figure out how to test it!